api/secrets.proto

```diff
@@ -0,0 +1,106 @@
+syntax = "proto3";
```

Comment:
As suggested by @stevvooe, it'd probably be good to list the pros and cons of having a single control API for everything, vs separate control APIs for each type of subsystem:

Everything in 1 monolithic control API protocol:
1.  Pro: Organization is simple, and anyone looking at this one protocol can see what can or cannot be changed about the cluster
1.  Pro: It's easy to see and enforce consistency across function arguments and naming

Control APIs split up across subsystems:
1.  Pro:  Downstream libraries can more tightly specify their dependencies: for example, they may need only a subset of behavior rather than entire control API.  For example, maybe they may depend only on secrets, and could be provided with something that provides the the same secrets API rather than all the whole control API (e.g. providing some sort of secrets management for your build time secrets on your build machine).
1.  Pro:  As @stevvooe suggests it's easier to version the control APIs for different components.  So if the API needed to change for a single component, we could have 2 different version implementations for just that component, as opposed to the whole control API.
1.  Pro: As @stevvooe suggests, it may allow users to switch to different implementations of the different subcomponents?  (e.g. volumes or networks, or even secret backends).
1.  Con: None of this actually exists yet, so do we want to do this now as opposed to deal with splitting it up later?

(I've probably missed some, just jotted some stuff down from convo earlier)

Are we ever going to want to support more modularity in terms of the control of components?  (e.g. differentially versioning certain components, swapping out implementations, providing partial implementations?)  If so, maybe it's worthwhile to start splitting some stuff up now?